crates/typos-cli/src/config.rs

```diff
@@ -455,15 +485,16 @@ impl DictConfig {
     }
 }
 
-fn find_project_file(dir: &std::path::Path, names: &[&str]) -> Option<std::path::PathBuf> {
+fn find_project_files(dir: &std::path::Path, names: &[&str]) -> Vec<std::path::PathBuf> {
     let mut file_path = dir.join("placeholder");
+    let mut found_files = vec![];
     for name in names {
         file_path.set_file_name(name);
         if file_path.exists() {
-            return Some(file_path);
+            found_files.push(file_path.clone());
```

Comment:
With this new structure, there is no point in creating the mutable `file_path` but we can instead just `found_files.push(dir.join(name))`